### PR TITLE
Add `rewrite` to the API

### PR DESCRIPTION
**Pull request by ajreynol**
This command is similar to `simplify` but does not apply top-level substitutions.

### CODE REVIEW

File: src/api/python/cvc5.pxi [hunk 1]
```diff
@@ -1726,6 +1726,24 @@ cdef class Solver:
         term.cterm = self.csolver.simplify(t.cterm)
         return term
 
+    def rewrite(self, Term t):
+        """
+            Rewrite a formula.
+        
+            This performs theory normalization on t and returns the result. In 
+            contrast to simplify, the current assertions are not used. The
```
Comment (aniemetz) [inline src/api/python/cvc5.pxi:1734]: Ditto (above)

File: src/parser/api/cpp/command.cpp [hunk 2]
```diff
@@ -1245,13 +1245,23 @@ void DeclareHeapCommand::toStream(std::ostream& out) const
 /* class SimplifyCommand                                                      */
 /* -------------------------------------------------------------------------- */
 
-SimplifyCommand::SimplifyCommand(cvc5::Term term) : d_term(term) {}
+SimplifyCommand::SimplifyCommand(cvc5::Term term, bool doSimplify)
```
Comment (aniemetz) [inline src/parser/api/cpp/command.cpp:1248]: Wouldn't it be cleaner to have an explicit `RewriteCommand`?

File: src/smt/solver_engine.cpp [hunk 3]
```diff
@@ -992,6 +992,18 @@ Node SolverEngine::simplify(const Node& t)
   return ret;
 }
 
+Node SolverEngine::rewrite(const Node& t)
+{
+  beginCall(true);
+  // now rewrite
+  Node ret = d_env->getRewriter()->rewrite(t);
+  // make so that the returned term does not involve arithmetic subtyping
```
Comment (aniemetz) [inline src/smt/solver_engine.cpp:1000]:
```suggestion
  // ensure that the returned term does not involve arithmetic subtyping
```

File: src/smt/solver_engine.h [hunk 4]
```diff
@@ -471,12 +471,16 @@ class CVC5_EXPORT SolverEngine
    * definitions, assertions, and the current partial model, if one
    * has been constructed.  It also involves theory normalization.
    *
-   * @throw TypeCheckingException, LogicException
+   * This returns a term that is equivalent to t in all models of the current
+   * assertions.
+   */
+  Node simplify(const Node& t);
+  /**
+   * Rewrite the given term, initializing if necessary.
```
Comment (aniemetz) [inline src/smt/solver_engine.h:479]: `initializing` in what sense?

**Review by aniemetz — COMMENTED** [review]
Note: Some of my suggested doc changes will require reformatting.

**Comment by ajreynol** [comment]
In a previous meeting, there was a suggestion to make a `uint_32 level` to determine what level of rewriting, e.g. `1=basic, 2=extended ...`. I think I will follow this idea.

**Pull request closed without merging**